src/bootstrap/builder.rs

```diff
@@ -437,7 +437,8 @@ impl<'a> Builder<'a> {
              compiler: Compiler,
              mode: Mode,
              target: Interned<String>,
-             cmd: &str) -> Command {
+             cmd: &str,
+             rustdoc_libdir: Option<PathBuf>) -> Command {
```

Comment:
Hm isn't this argument always `builder.rustc_libdir(compiler);` basically? If so, could this avoid being passed as an argument and just be set internally?

Reply:
No, `compiler` here could be stage1 if `Build::force_use_stage1` returns true but real launched compiler will be from stage2

Reply:
Indeed yeah but rustdoc doesn't work at all in stage 1, so I think this is always the stage2 version for rustdoc?

Reply:
Yes, so real stage2 `rustdoc` binary is choosen with `REAL_RUSTDOC` environment variable while all other parameters from stage1.

Reply:
Er so my point is that this argument is a constant, it's always generated from a stage2 compiler. In that case we can just inline it here, right?

Reply:
Not sure. What if build option `build.full-bootstrap` will be enabled? Do we have to provide path to stage3 libs?
Also, `Builder::cargo` could be called from other stages also.

Reply:
I've tried to do it with 
```rust
cargo.env("RUSTDOC_LIBDIR", self.rustc_libdir(self.compiler(2, self.build.build)));
```
but it fails with error:
```
thread 'main' panicked at '

Cycle in build detected when adding Assemble { target_compiler: Compiler { stage: 2, host: "x86_64-unknown-linux-gnu" } }
        Any
        Any
        Any
        Any
        Any
', bootstrap/builder.rs:673:17
```

Reply:
Hm, does the same error happen if you only set this env var when the command is `"doc"`?

Reply:
I'll try it but it also required here https://github.com/o01eg/rust/blob/50a80028052a4ca4ad939e926032bce8ac4d3cdd/src/bootstrap/check.rs#L1178
Are there "test" command?

Reply:
Sure yeah, maybe set it for everything that's not `"build"`?

Reply:
Yep, it helped.